vllm/v1/executor/multiproc_executor.py

```diff
@@ -234,6 +289,22 @@ def take_draft_token_ids(self) -> DraftTokenIds | None:
             "take_draft_token_ids", unique_reply_rank=self.output_rank
         )
 
+    def get_response_mqs(
+        self, unique_reply_rank: int | None = None
+    ) -> list[MessageQueue]:
+        message_queues = []
+        for rank in range(self.world_size):
+            if rank < self.local_world_size:
+                local_message_queue = self.workers[rank].worker_response_mq
+                message_queues.append(local_message_queue)
+            else:
+                remote_message_queue = self.workers[0].rpc_response_mqs[rank]
+                assert remote_message_queue is not None
+                message_queues.append(remote_message_queue)
+        if unique_reply_rank is not None:
+            message_queues = [message_queues[unique_reply_rank]]
+        return message_queues
```

Comment:
We should avoid constructing this list each time, can set `self.worker_response_mqs = self._get_response_mqs()` near the end of `_init_executor()` (suggest right before `start_worker_monitor()`. Without unique reply rank of course, that part can stay equivalent to how it already was below...